src/cms/views/events/event_view.py

```diff
@@ -191,7 +191,9 @@ def post(self, request, **kwargs):
             if event_form.cleaned_data.get("is_recurring"):
                 # If event is recurring, save recurrence rule
                 event_form.instance.recurrence_rule = recurrence_rule_form.save()
-            elif event_form.fields["is_recurring"].has_changed():
+            elif event_instance and event_form.fields["is_recurring"].has_changed(
+                event_instance.is_recurring, event_form.cleaned_data.get("is_recurring")
+            ):
                 # If the event is not recurring but it was before, delete the associated recurrence rule
                 event_form.instance.recurrence_rule.delete()
```

Comment:
```suggestion
                event_form.instance.recurrence_rule.delete()
                event_form.instance.recurrence_rule = None
```
I still get the error `save() prohibited to prevent data loss due to unsaved related object 'recurrence_rule'.` when I change an event from recurring to non-recurring. I'm not completely sure what causes the error, but if I understood it correctly, it's the following:
- The `delete()` deletes the recurrence rule in the database, but it does not change the recurrence rule of the event instance
- The event instance still has the recurrence rule as before, with the only difference being that the recurrence rule is not in the database anymore
- Django sees that you want to save the event which has an unsaved foreign key object and throws the error

So I presume that we need both: deleting the recurrence object (so the database doesn't get polluted with old and unused recurrence rule objects) and to set the recurrence rule of the event explicitly to `None`.

Reply:
That is absolutely right, I totally missed that part.
Since I was only looking at the creation of new events according to the linked issue, this problem absolutely slipped through and I didn't test for this functionality.